### DISCUSSION TITLE
Communication Problems

### DISCUSSION THREAD

**Discussion by T-C123** [jgyates/genmon#1223] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

Latest

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Sorry I can't get all system details due to the problem I'm having.

I got an email message that my Generac 18 kW generator was exercising and I opened my browser to see the status. I was unable to connect to the generator. IP address was correct, router shows no problems. I'm still getting emails and can send email commands to Genmon to request a status report. Just not able to see the Genmon screen. Tried different browsers and VNC - no help. Running on a Raspberry Pi 4.

It's about 10 degrees out now so I won't be able to look further until the weather warms up, but what should I be looking for?

Thanks much!

To

**Comment by jgyates** [comment]
You mentione that you can not connect to the browser. What response do you get in the browser? timeout, blocked, etc?

Can you ping the pi? 
Are you using a static or dynamic IP address? 
Do you have an ASUS router (they are known to block some local web servers)?

**Reply by jgyates** [reply]
Can you access you pi via VNC or SSH?

**Comment by T-C123** [comment]
Browser returnc "The connection has timed out"
Ping returns "Destination host unreachable"
I'm using a static IP address on a TP-Link router. It has been working without issues for months then lost communication.
VNC returns "Timed out waiting for a response from the computer"
SSH returns "Connection timed out"

Thanks!

**Reply by skipfire** [reply]
That sounds like the OS has probably just completely hung, potentially micro-SD failure or it just locked up.  If all the connection options are timing out you are probably stuck with needing to power cycle it.

**Reply by T-C123** [reply]
I'm hesitant to try a power cycle without seeing what is going on. I don't want to overwrite anything that might give a clue to what happened and how to prevent it in the future. Waiting for better weather (isn't that always the case when things go wrong?) to connect a monitor and see.

If the OS hung, why would I still be receiving change of status emails and be able to send email commands to Genmon? Seems like it must be something more specific.

Thanks!

**Reply by skipfire** [reply]
If it was something specific, then most stuff would work, but with ssh, vnc, genmon's UI all not working that indicates it's a non-specific problem and fairly global.

I don't think anything would get overwritten, though current state would get lost.  While not the same and not genmon, I have a Pi 5 with an multi-NVMe adapter board that the NVMe controller reset occasionally and when it reset anything that was actively in memory still worked, but nothing not in memory failed, I couldn't even log in directly on the console.  I had an open SSH session in one instance and all commands (including `ls` and `reboot`) reported not found.  The core of Raspberry Pi OS does load into memory, as does a notable chunk of genmon (because of how Python works), but not everything is in memory (such as page templates).

**Reply by jgyates** [reply]
as a test you could change the generator from auto to off, wait about 15 seconds and then set it back to auto. If the pi is still up and genmon operational, you should get emails on the state change of the generator.

If you don't get the emails and you can log in via ssh then the only recourse would be to reboot.

**Reply by jgyates** [reply]
I agree with @skipfire  about the SD card. It also could be a power / ground issue.

**Reply by T-C123** [reply]
I am getting status updates by email as well as being able to send email commands ("generator: status") to Genmon. SSH & VNC not working. Waiting for the snow to stop.

**Reply by T-C123** [reply]
I sent log and register files in case you want to see what's going on.

**Reply by jgyates** [reply]
I do not see anything that would cause a system disruption. There are a couple of errors but they deal with MQTT input add on and an error related to not loading a gauge that may be related to that. Check to make sure all of your external sensors are working (temp or anything imported as an input to genmon via MQTT. Other that that, everything looks good.

**Comment by T-C123** [comment]
Working now! Walking past my router and on a whim I decided to power off, reboot. When it came back on communication with the Pi was restored. I don't know if the reboot fixed something in the router or triggered the Pi to reset. Previously, all settings in the router looked good and it showed a reserved, permanent IP address for the Pi. All other connections through the router were working normally. Still don't know what caused the issue.

Thanks to everyone for their help and suggestions!

**Reply by skipfire** [reply]
That brings 2 likely problem causes that can be fixed with the same thing.

If the router is also your wifi, then the Pi likely dropped wifi and reacquired which triggered a DHCP call and refresh.  Your Pi may have somehow gotten the wrong IP at some point.

Second (and I'm guessing more likely), perhaps the name resolution or the routing tables in the router was borked and rebooting cleared garbage out of RAM and rebuilt correctly when it came back up.

In either case, I'm glad you are working again, especially without having had to touch anything on the Pi.

**Reply by T-C123** [reply]
Still thinking on this and wondering if there is a way that the IP address could still be active allowing email but the port 8000 access could be disabled preventing SSH, VNC, or browser access to Genmon? Would port 8000 reconnect on reboot of the router and/or reboot of the Pi?
Thoughts?

Thanks!

**Reply by skipfire** [reply]
A reboot of the router would not impact the Pi listening on SSH, VNC, or 8000 and there's no real link between any of those as they are all independent services.  Ping is the most basic of network services as well and your Pi wasn't responding to that either.
